src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -309,6 +310,61 @@ object Task {
     private val LegacyTaskIdRegex = """^(.+)([\._])([^_\.]+)$""".r
     private val ResidentTaskIdRegex = """^(.+)([\._])([^_\.]+)(\.)(\d+)$""".r
 
+    // UUID Parse according to [[UUID.toString]]
+    def uuidP[_: P]: P[UUID] = P((TimeLow ~ "-" ~ TimeMid ~ "-" ~ TimeHighAndVersion ~ "-" ~ VariantAndSequence ~ "-" ~ Node).!).log.map(UUID.fromString(_))
+
+    def TimeLow[_: P] = P(HexOctet.rep(exactly = 4))
+    def TimeMid[_: P] = P(HexOctet.rep(exactly = 2))
+    def TimeHighAndVersion[_: P] = P(HexOctet.rep(exactly = 2))
+    def VariantAndSequence[_: P] = P(HexOctet.rep(exactly = 2))
+    def Node[_: P] = P(HexOctet.rep(exactly = 6))
+    def HexOctet[_: P] = P(HexDigit ~ HexDigit)
+    def HexDigit[_: P] = P(CharIn("0-9a-fA-F"))
+
+    // "^(
+    //    ( [a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9] )\\.)*([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9])|(\\.|\\.\\.
+    //   )$".r
+    // app.test.23.b6ff5fa5-7714-11e7-a55c-5ecf1c4671f6.41
+    def safeRunSpecIdP[_: P]: P[AbsolutePathId] = P(CharsWhileIn("a-z0-9_.").!).map(PathId.fromSafePath(_)).log
```

Comment:
This is where it currently fails. The `.` must have precedence to be a separator.